crates/network-scanner/examples/boardcast.rs

```diff
@@ -0,0 +1,35 @@
+use std::{process::exit, time::Duration};
+
+use network_scanner::boardcast::boardcast;
+use tokio_stream::StreamExt;
+
+#[tokio::main]
+pub async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::SubscriberBuilder::default()
+        .with_max_level(tracing::Level::TRACE)
+        .init();
+
+    let ip = std::net::Ipv4Addr::new(192, 168, 1, 255);
+
+    let mut iterator = boardcast(ip, Some(Duration::from_secs(1))).await?;
+
+    while let Some(result) = iterator.next().await {
+        match result {
+            Ok(res) => {
+                tracing::info!("received result {:?}", &res)
+            }
+            Err(e) => {
+                if let Some(e) = e.downcast_ref::<std::io::Error>() {
+                    // if is timeout, say timeout then break
+                    if let std::io::ErrorKind::TimedOut = e.kind() {
+                        tracing::info!("timed out");
+                        exit(0)
+                    }
+                }
```

Comment:
question: Do we need to specifically handle the timeout case?

Reply:
yes, please see the above comments